### PR TITLE
Implemented intraday and historical stock price endpoints

### PR DESCRIPTION
**Pull request by avidlaud**
Created the stock service endpoints for requesting prices. Addresses #15

# Endpoints
## Intraday:
`{url}/equity/intraday/:equityTicker`

Usage:
Intraday prices for Twitter (twtr):
`{url}/equity/intraday/twtr`

Return body:
Object of time:average_price pairs
image

## Historical:
`{url}/equity/historical/:equityTicker/?timeframe=`
Allows arguments of: `'5d', '1m', '6m', 'ytd', '1y',  or '5y'`

Usage:
1 month prices for Twitter (twtr):
`{url}/equity/historical/twtr/?timeframe=1m`

Return body:
Object of date:close_price pairs

image

# Caching
The variable `CACHE_TIME` in `stockController.js` defines the amount of time that an entry is considered recent enough. The value is in minutes.

### CODE REVIEW

**Review by christinem61 — DISMISSED** [review]

File: app/services/stock/controllers/stockController.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            { ticker: equityTicker },
+            { $set: { 'intraday.prices': priceData }, $currentDate: { 'intraday.lastUpdated': true } },
+            {
+                new: true,
+                upsert: true,
+            },
+        );
+        return updatedEquity.intraday;
+    } catch (err) {
+        console.error(err);
```
Comment (JScheire) [inline app/services/stock/controllers/stockController.js:44]: can we send as a response with appropriate status code

File: app/services/stock/controllers/stockController.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            { ticker: equityTicker },
+            { $set: { [`${timeFrame}.prices`]: priceData }, $currentDate: { [`${timeFrame}.lastUpdated`]: true } },
+            {
+                new: true,
+                upsert: true,
+            },
+        );
+        return updatedEquity[timeFrame];
+    } catch (err) {
+        console.error(err);
```
Comment (JScheire) [inline app/services/stock/controllers/stockController.js:78]: same thing here

**Comment by avidlaud** [comment]
I'll address the error handling issues tomorrow; will also implement a `/equity/current/:ticker` endpoint for easier order and portfolio calculations.

**Comment by avidlaud** [comment]
Added some basic error handling - it's not very descriptive but all calls are requests are made internally in our application so it shouldn't be much of an issue.

Implemented `/equity/current/:ticker` endpoint for easy current price access.

**Review by RiyaTayal20 — DISMISSED** [review]

**Review by JScheire — DISMISSED** [review]

**Review by aarushipandey — APPROVED** [review]

**Review by sahigande — DISMISSED** [review]

**Comment by avidlaud** [comment]
I think this should now cleanly merge

**Review by christinem61 — APPROVED** [review]

**Review by RiyaTayal20 — APPROVED** [review]

**Review by aarushiS — APPROVED** [review]

**Pull request merged**